src/main/java/org/testng/internal/thread/graph/GraphThreadPoolExecutor.java

```diff
@@ -54,6 +60,9 @@ public GraphThreadPoolExecutor(
   public void run() {
     synchronized (m_graph) {
       List<T> freeNodes = m_graph.getFreeNodes();
+      if (m_comparator != null) {
+          Collections.sort(freeNodes, m_comparator);
```

Comment:
The sort logic could be done in `graph.getFreeNodes()`. The comparator could be set by constructor or maybe setter.